sre_maturity.adoc

```diff
@@ -0,0 +1,164 @@
+
+The SLO might only be one around service availability and it may not even be formally tracked at the start of this phase (as one is first learning to walk). Towards the end of the phase, as the team transitions to "running," the they may start to wonder how to enforce that availability SLO and other new ones they want to put into place. 
+
+There's a graduation process with SLOs from "informing" to "enforcing". You can use SLOs to inform decisions in this early stage. As they get refined and wider awareness & support, you can think about enforcement of an error budget policy.
+
+
+=== Phase 2: The 'Walk' Phase
+
+You have established patterns and guidelines for observability signals such as metrics, logs, errors etc...
+You are using these patterns to speed up debugging of issues and correlating between signals easier, thereby reducing your Mean Time To Recovery (MTTR).
+Your alerts have matured and rarely give false positives as a result of having these patterns and refining queries.
+
+* SLOs - SLOs have been iterated on, and you have more than 1 meaningful SLO.
+* You understand and track error budgets
+* Actionable Alerts
+* Self healing
+  * Avoid SRE having sub-systems that automagically work around the inefficiencies of upstream - needs a good argument why it can’t be fixed upstream - pushing left
+* Combination of internal (whitebox) and external (blackbox) monitoring - tied to SLOs
+* Dashboards, logging (easily accessible)
+* Improving the architecture for lower cost to serve. Pereq here - calculating & knowing the cost to serve
+
+=== Phase 3: The 'Run' Phase
+
+* Observability correlation beteween logs, metrics, alerts, events etc..
+* build on this foundation looking at AI Ops and improving the debug experience (focused on decreasing MTTR further)
+  * Needs everything underneath it first (all the signals)
+* SLOs - Well established SLO Review process
+
+
+== On Call, Escalations and Incidents
+
+=== Phase 1: The ‘Crawl’ Phase
+
+You have a team with an on call schedule.
+It doesn’t have to be 24/7. It doesn’t have to be automated.
+But there must be someone responsible to respond when things go bad.
+Setting expectations at this early stage with a reduced SLO means you can tailor the schedule to more reasonable times for your team.
+
+Following on from this, you have an escalation process. A way to be notified of problems.
+It should be possible to raise a problem with the SRE team and the people that build the service, like engineers, QE, writers.
+This can be a mailing list, slack channel, issue system, whatever.
+Formalise it and share it.
+
+If things go really bad, you have an Incident Management Process and do a Root Cause Analysis (RCA) (sometimes called a Post Mortem or Post Incident Review).
```

Comment:
There's a whole class of Incident Review processes, and some companies really love one over the other.  What about reorganizing the next few bullets a bit:

If things go really bad, you have a documented Incident Management Process that people follow.

Within a few days after incidents are resolved, while things are still fresh in your minds, the people involved have a 'blameless' review to record what happened and learn from the incident.  Some examples of review and learning processes are Post Mortems, Post Incident Reviews, Learning Reviews, and Root Cause Analysis (RCA).

You should start to build up a library of incident reviews that can feed into doing more proactive things as you mature e.g. Incident rehearsals.

Each incident review feeds *extremely valuable*... etc.

Reply:
We're soon going to have a dedicated workstream for Incident Management, so I think the type of content you're talking about here will be developed there.  For the sake of this maturity doc having a process of any kind is part of the culture+mentality we are trying to measure, and I think should be good enough?